js/packages/teams-ai/src/AIApiFactory.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * @module teams-ai
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TurnState } from './TurnState';
+import { DefaultTurnState } from './DefaultTurnStateManager';
+import { PromptTemplate } from './Prompts';
+import { ConfiguredAIOptions } from './AI';
+import { TurnContext } from 'botbuilder';
+import { Plan } from './Planner';
+import { ResponseParser } from './ResponseParser';
+
+export interface AIApiFactoryOptions {
+    /**
+     * Optional. A flag indicating if the planner should only say one thing per turn.
+     * @summary
+     * The planner will attempt to combine multiple SAY commands into a single SAY command when true.
+     * Defaults to false.
+     */
+    oneSayPerTurn?: boolean;
+}
+
+export interface AIApiFactory<TState extends TurnState = DefaultTurnState> {
+    completePrompt(
+        context: TurnContext,
+        state: TState,
+        prompt: PromptTemplate,
+        options: ConfiguredAIOptions<TState>
+    ): Promise<string>;
+
+    generatePlan(
+        context: TurnContext,
+        state: TState,
+        prompt: PromptTemplate,
+        options: ConfiguredAIOptions<TState>
+    ): Promise<Plan>;
+}
+
+/**
+ *
+ * @template TState
+ * @param {string} response The response from the OpenAI API.
+ * @param {ConfiguredAIOptions<TState>} options The options for the AI configuration.
+ * @param {boolean} oneSayPerTurn * Optional. A flag indicating if the planner should only say one thing per turn.
+ * @returns {Plan} The plan that was generated.
+ */
+export function createPlan<TState extends TurnState = DefaultTurnState>(
```

Comment:
It seems like the contents of this file related more to plans rather than the Open AI API itself, would it be better to rename this class?

Reply:
Not sure what you mean here. This isn't a class, just a helper function applicable to any service. The intent is that someone implementing their own factory could import this function and use it.
This file is not related to OpenAI, it's related to providing the Plan contracts to the SDK. Let me know if you have a different name suggestion based on the above.